enhancements/olm/operator-must-gather.md

```diff
@@ -0,0 +1,137 @@
+
+## Summary
+
+Components developing [must gather images]("../oc/must-gather.md#must-gather-images") to gather all the information that they will need for debugging has become the defacto mechanism for customers to father information for a support case. When debugging installation of an operator with OLM we need to gather data from objects across the cluster. 
+
+## Motivation
+
+Today back and forth between the customer and the operator team or the OLM team to get info about an operator failing install. This the current state or each operator needs to write their own must gather to capture everything. Operator authors should have a starting point for writing their own must gathers such that gathering the OLM information for their operator should be consistent. 
+
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+1. Gathering should gather exactly the operator, and relatedObjects for that operator, and the resources created by OLM.
+2. A library that can be used to enable this functionality across multiple commands.
+3. A small set of user facing CLI flags. 
+4. Integaration in a seperate must-gather image for other operator authors, or via the library should
+5. Should be 100% functional when not in a openshift environment.
+
+### Non-Goals
+
+1. Anything related to the wider cluster. This is not meant to be a OCP debug tool.
+2. Specficics for a given operator. This is meant to help debug install of the operator, not the operator's functionality. 
+3. Handle low level install scenarios, such as manually creating a CSV after applying RBAC.
+
+## Proposal
+
+A library, in the [api](https://github.com/operator-framework/api/tree/master/pkg) repo that will be vendored by `OLM` and `operator-sdk`. This library will include all of the traversal logic for retrieving OLM Operators data. 
+
+The library will include a set of pflags that can be used by commands to create the options structure.
+
+OLM will create a must-gather image and productize it. Allowing customers to run this must gather when needed.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Constraints
+This enhancement is only meant for operators that are deployed by OLM. The entrypoint should be the `subscription`.
+
+### Risks and Mitigations
+- No environment variables will be outputed from a pod.
+- We will not save any secret data that may have been created. 
+
+## Design Details
+* Considering that many different CLI's will be interacting with the library, the library will create an interface that the CLI's will need to implement to write files to disk.
+* The library will have a default version of this interface internally
+
+```
+type ObjectWriter interface {
+  // object is the object to be written. The implmenter can choose the output format.
+  // file is the relative path, from the starting directory of the inspect command, for the object.
+  Write(object interface{}, file string) error
```

Comment:
Why not just have an `io.Writer` (or some appropriate variant)? is it critical that consumers specify a filesystem path?